### PR TITLE
Adding downloadMessage as a prop

### PR DESCRIPTION
**Pull request by mmathewsTableau**
Adding downloadMessage as a prop to react native web view so that the library consumer can specify the message shown in when downloading content.

Meant to address this issue: https://github.com/react-native-webview/react-native-webview/issues/2436

For our app, this allows the message to be localized (see the toast near the bottom):

### CODE REVIEW

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java [hunk 1]
```diff
@@ -95,25 +95,27 @@ private enum MimeType {
...
+      public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+          case FILE_DOWNLOAD_PERMISSION_REQUEST: {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+              if (downloadRequest != null) {
+                downloadFile(downloadMessage);
+              }
+            } else {
+              Toast.makeText(getCurrentActivity().getApplicationContext(), "Cannot download files as permission was denied. Please provide permission to write to storage, in order to download files.", Toast.LENGTH_LONG).show();
```
Comment (TheAlmightyBob) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java:110]: Oh dang.... somehow I missed that this string was also hardcoded. 😞 

Probably need the same solution here... although it could be a separate PR... and boy this is a much more complicated string to have to localize...

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 2]
```diff
@@ -159,6 +159,7 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
   protected boolean mAllowsFullscreenVideo = false;
   protected @Nullable String mUserAgent = null;
   protected @Nullable String mUserAgentWithApplicationName = null;
+  protected String mDownloadMessage = "Downloading";
```
Comment (TheAlmightyBob) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:162]: I'd suggest making "Downloading" a string constant, so for example it could still be used if a custom string was specified and then reset back to `undefined`? (though admittedly I don't know how likely that is... just seems like it would be a small change to be extra safe?)

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 3]
```diff
@@ -287,6 +288,12 @@ public void setShowsVerticalScrollIndicator(WebView view, boolean enabled) {
     view.setVerticalScrollBarEnabled(enabled);
   }
 
+  @ReactProp(name = "downloadMessage")
```
Comment (TheAlmightyBob) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:291]: Do you think `downloadingMessage` might be clearer, just to avoid any possible confusing about it being a "download success" message...?

(I know the original variable name was `downloadMessage`, but since that was just a local variable in the function there was less likely to be any confusion)

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 3 above)
Comment (mmathewsTableau) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: changed to downloadingMessage

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 2 above)
Comment (mmathewsTableau) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Created a separate variable for default strings

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 4]
```diff
@@ -248,15 +253,23 @@ public void onDownloadStart(String url, String userAgent, String contentDisposit
...
+        if (module.grantFileDownloaderPermissions(getDownloadingMessage(), getLackPermissionToDownloadMessage())) {
+          module.downloadFile(getDownloadingMessage());
         }
       }
     });
 
     return webView;
   }
 
+  private String getDownloadingMessage(){
```
Comment (TheAlmightyBob) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:265]: ultra-nit, but it would be good to follow the style standard of the rest of the file, which would mean adding a space before the bracket

(as well as updating the setters so that the opening bracket is on the same line as the function signature)

(also it's slightly weird to have the getters separated from the setters, but I totally get that to do otherwise would break the grouping of React props, so I don't have a better idea, just noting that quirk)

Comment (mmathewsTableau) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Fixed the spacing and placement of the function brackets

**Review by TheAlmightyBob — APPROVED** [review]
